homeassistant/components/zwave/__init__.py

```diff
@@ -600,14 +601,36 @@ def start_zwave(_service_or_event):
     return True
 
 
-class ZWaveDeviceEntity:
+class ZWaveDeviceEntity(Entity):
     """Representation of a Z-Wave node entity."""
 
     def __init__(self, value, domain):
         """Initialize the z-Wave device."""
+        # pylint: disable=import-error
+        from openzwave.network import ZWaveNetwork
+        from pydispatch import dispatcher
         self._value = value
         self.entity_id = "{}.{}".format(domain, self._object_id())
 
+        dispatcher.connect(
+            self.network_value_changed, ZWaveNetwork.SIGNAL_VALUE_CHANGED)
+
+    def network_value_changed(self, value):
+        """Called when a value has changed on the network."""
+        if self._value.value_id == value.value_id or \
+           self._value.node == value.node:
+            _LOGGER.debug('Value changed for label %s', self._value.label)
+            self.value_changed(value)
+
+    def value_changed(self, value):
+        """Called when a value for this entity's node has changed."""
+        self.update_properties()
+        self.schedule_update_ha_state()
+
+    def update_properties(self):
```

Comment:
I don't believe `update` is strictly for polling updates. I think it would be more coherent and cleaner to use it here too. Then you would only need to call `schedule_update_ha_state(True)` in the callback for push updates. I use this pattern for mysensors.